tests/test_merge_streams_block.py

```diff
@@ -160,3 +160,21 @@ def test_reset_expiration_job_on_new_signal_input_2(self):
         sleep(0.05)
         blk.stop()
         self.assertEqual(blk._signal_expiration_job.call_count, 0)
+
+    def test_recursive_case(self):
+        """ Two signals with dictionary values recursively merge together"""
+        blk = MergeStreams()
+        blk._signal_expiration_job = MagicMock()
+        self.configure_block(blk, {
+            "expiration": {"seconds": 0.1},
+            "notify_once": False
+        })
+        blk.start()
+        blk.process_signals([Signal({"key": {"inner_key_1": 1}})],
+                            input_id="input_1")
+        blk.process_signals([Signal({"key": {"inner_key_2": 2}})],
+                            input_id="input_2")
+        blk.stop()
+        self.assert_num_signals_notified(1)
+        self.assertDictEqual(self.last_notified[DEFAULT_TERMINAL][0].to_dict(),
+                             {"key": {"inner_key_1": 1, "inner_key_2": 2}})
```

Comment:
This test looks good, but what happens when you have more than one attribute in your signal? Add a test (or just modify this one) so that your signals have more attributes than just `key`.

Or what about when both signals have `key` but one is a dict and one is a string? This would make more another good test.